pkg/cmd/server/api/types.go

```diff
@@ -47,10 +47,14 @@ type MasterConfig struct {
 	// CORSAllowedOrigins
 	CORSAllowedOrigins []string
 
+	// MasterPublicURL is how clients can access the OpenShift API server
+	MasterPublicURL string
```

Comment:
Oh, and for the next question: each stanza has its own copy to keep it contained if we decide to allow starting these components separately.